portfolio/src/main/java/com/google/sps/servlets/DeleteCommentsServlet.java

```diff
@@ -27,19 +27,27 @@
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
 
-/** Servlet responsible for deleting all comments data. */
-@WebServlet("/delete-data")
-public class DeleteDataServlet extends HttpServlet {
+/** Servlet responsible for deleting one comment if an id is passed, or otherwise deleting all comments. */
+@WebServlet("/delete-comments")
+public class DeleteCommentsServlet extends HttpServlet {
   @Override
   public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
     DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
-    Query query = new Query("Comment");
-    PreparedQuery results = datastore.prepare(query);
 
-    for (Entity commentEntity : results.asIterable()) {
-      long id = commentEntity.getKey().getId();
+    String idString = request.getParameter("id");
+    if (idString == null) {
+      Query query = new Query("Comment");
+      PreparedQuery results = datastore.prepare(query);
+
+      for (Entity commentEntity : results.asIterable()) {
+        long id = commentEntity.getKey().getId();
+        Key commentEntityKey = KeyFactory.createKey("Comment", id);
+        datastore.delete(commentEntityKey);
+      }
+    } else {
+      long id = Long.parseLong(idString);
```

Comment:
you've already defined id above on line 43. you can just reassign the value here.
```suggestion
      id = Long.parseLong(idString);
```

Reply:
But line 43 is in the if-block, so I think I need to define id again in the else-block?

Reply:
Ties in to what looked misaligned back then. ignore.